staging/src/kubevirt.io/client-go/kubecli/handler.go

```diff
@@ -210,65 +214,53 @@ func (v *virtHandlerConn) Pod() (pod *v1.Pod, err error) {
 	return v.pod, err
 }
 
-func (v *virtHandlerConn) Put(url string, tlsConfig *tls.Config, body io.ReadCloser) error {
+func (v *virtHandlerConn) doRequest(req *http.Request) (response string, err error) {
+	resp, err := v.httpClient.Do(req)
+	if err != nil {
+		return
+	}
 
-	client := http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: tlsConfig,
-		},
-		Timeout: 10 * time.Second,
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return "", fmt.Errorf("unexpected return code %s", resp.Status)
 	}
 
-	req, err := http.NewRequest(http.MethodPut, url, body)
+	defer resp.Body.Close()
+	responseBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return err
+		return "", fmt.Errorf("cannot read get body %s", resp.Status)
 	}
 
-	resp, err := client.Do(req)
+	response = string(responseBytes)
+	return
+}
+
+func (v *virtHandlerConn) Put(url string, body io.ReadCloser) error {
+	req, err := http.NewRequest(http.MethodPut, url, body)
 	if err != nil {
 		return err
 	}
 
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		return fmt.Errorf("unexpected return code %s", resp.Status)
+	_, err = v.doRequest(req)
+	if err != nil {
+		return err
```

Comment:
nit: you can simply return `err` here

Reply:
Although you're logically right, I actually prefer this style. It prepares the ground for future addition to the function and clearer IMO